src/common/raft_metric.h

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) 2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef COMMON_RAFT_METRIC_H
+#define COMMON_RAFT_METRIC_H
+
+#include <algorithm>
+#include <string>
+#include <unordered_map>
+
+#include "raft/distance/distance_types.hpp"
+#include "knowhere/comp/index_param.h"
+#include "knowhere/expected.h"
+
+namespace knowhere {
+
+inline expected<raft::distance::DistanceType, Status>
+Str2RaftMetricType(std::string metric) {
+    static const std::unordered_map<std::string, raft::distance::DistanceType> metric_map = {
+        {metric::L2, raft::distance::DistanceType::L2Expanded},
+        {metric::IP, raft::distance::DistanceType::InnerProduct},
+        {metric::HAMMING, raft::distance::DistanceType::HammingUnexpanded},
```

Comment:
Not sure if it makes a difference here or if Knowhere will just silently skip in the case of an exception but I don't think we support hamming or jaccard on the IVF APIs yet. 

Both of these can be formulated post processing on a dot product, though, so we might be able to support them in a future release.